src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +540,14 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual IdentityModel.Clients.ActiveDirectory.AuthenticationResult RedeemAuthorizationCode(string authorizationCode)
```

Comment:
Oh please no, no dependency on AAD stuff... specially for such a simple thing.
Why not simply using a good old `HttpClient`?

https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/samples/Mvc/Mvc.Client/Startup.cs#L76-L103